src/operator.rs

```diff
@@ -850,6 +851,41 @@ impl UnaryPreAggFunction for ParseJson {
     }
 }
 
+#[derive(Clone)]
+pub struct ParseLogfmt {
+    input_column: Option<Expr>,
+}
+
+impl ParseLogfmt {
+    pub fn new(input_column: Option<String>) -> ParseLogfmt {
+        ParseLogfmt {
+            input_column: input_column.map(Expr::Column),
+        }
+    }
+}
+
+impl UnaryPreAggFunction for ParseLogfmt {
+    fn process(&self, rec: Record) -> Result<Option<Record>, EvalError> {
+        let pairs = {
+            let inp = get_input(&rec, &self.input_column)?;
+            // Record includes the trailing newline, while logfmt considers that part of the
+            // message if present. Trim any trailing whitespace.
+            logfmt::parse(&inp.trim_end())
+        };
+        let res = {
+            pairs.iter().fold(Some(rec), |record_opt, pair| {
```

Comment:
You might as well do `pairs.into_iter()` which will consume while iterating (or just use a for loop and skip the fold). Then you won't need `to_string` and you can skip the copy.